### ISSUE TITLE
Sentence Transformers Wrapper for BioGPT- fine tuning

### ISSUE THREAD

**Issue by liel-blum** [UKPLab/sentence-transformers#2321]
Hi, I'm using sentence transformers with BioGPT as indicated  in #1824 
I want to fine tune it on the causal loss, which is already BioGPT's loss.

When fine-tuning BioGPT alone I can do it in this manner:
```
        collated_inputs = self.data_collator(inputs['input_ids'].tolist())
        collated_inputs = {k: v.to(self.device) for k, v in collated_inputs.items()}

        inputs['input_ids'] = collated_inputs['input_ids']
        inputs['labels'] = collated_inputs['labels']
        vals = self.gpt_model(**inputs)
        loss = vals.loss
```

Is there a way to access this loss from within the sentence transformers library?

**Comment by tomaarsen** [comment]
Hello!

Sentence Transformer works by finetuning with pairs or triplets & computing losses over their embeddings, whereas this is just training it for generation (presumably), which does not really make much sense. So, no, this loss isn't easily accessible with Sentence Transformers. If you want to finetune with causal loss, you can do so with normal fine-tuning via `transformers`.

I'll close this, as causal loss is not very related to Sentence Transformers. 

- Tom Aarsen
